Source/Blazorise/Components/Offcanvas/Offcanvas.razor

```diff
@@ -0,0 +1,13 @@
+﻿@namespace Blazorise
+@inherits BaseComponent
+<CascadingValue Value="@this" IsFixed>
+    <CascadingValue Value="@State">
```

Comment:
Very much needed. The state is a record, and records have equality by values instead of reference. And we can know when the parent state changes to rerender the child if needed. Without it makes, our lives are much more complicated.

Reply:
Even if you keep it because of the usefulness do you need **two** **CascadingValues**? Would just the **Parent** be enough instead?

Reply:
No, as I said. With only Parent how will we know that something in parent changed? We would have to have some other means of notification to the children. This is much easier.

For example this code in the child component.

```
[CascadingParameter]
protected OffcanvasState ParentOffcanvasState
{
    get => parentOffcanvasState;
    set
    {
        if ( parentOffcanvasState == value )
            return;

        parentOffcanvasState = value;

        DirtyClasses();
        DirtyStyles();
    }
}
```

Reply:
Checking in the parameters changed like I believe its recommended. Anyway I digress. It works...